go.mod

```diff
@@ -3,11 +3,35 @@ module github.com/grafana/dskit
 go 1.16
 
 require (
+	cloud.google.com/go v0.87.0 // indirect
+	github.com/HdrHistogram/hdrhistogram-go v1.1.0 // indirect
+	github.com/armon/go-metrics v0.3.6
 	github.com/go-kit/kit v0.11.0
+	github.com/gogo/googleapis v1.4.0 // indirect
+	github.com/gogo/protobuf v1.3.2
+	github.com/gogo/status v1.1.0
+	github.com/golang/snappy v0.0.4
+	github.com/gorilla/mux v1.8.0 // indirect
+	github.com/grpc-ecosystem/go-grpc-middleware v1.3.0
+	github.com/hashicorp/consul/api v1.9.1
+	github.com/hashicorp/go-cleanhttp v0.5.1
+	github.com/hashicorp/go-sockaddr v1.0.2
+	github.com/hashicorp/memberlist v0.2.3
+	github.com/matttproud/golang_protobuf_extensions v1.0.2-0.20181231171920-c182affec369 // indirect
+	github.com/miekg/dns v1.1.42 // indirect
 	github.com/pkg/errors v0.9.1
 	github.com/prometheus/client_golang v1.11.0
 	github.com/prometheus/common v0.26.0
 	github.com/stretchr/testify v1.7.0
+	github.com/thanos-io/thanos v0.22.0
```

Comment:
Surprised to see Thanos as dependency here... with little extra work, I think we could remove it (esp. DNS provider). But maybe that's better left for another PR.

Reply:
@pstibrany same here. I was going to ask if we can do what you're suggesting, i.e. extract what we need from it.

Reply:
I think biggest part would be DNS provider. Instead of direcly using one from Thanos, we can define an interface in dskit, provide basic implementation, and allow dskit clients to replace it with one from Thanos. But I would suggest to do that in a separate PR, together with copying/replacing other thanos dependencies (apart from DNS provider, I only see tiny usage from tests). WDYT?

Reply:
A separate PR is OK by me @pstibrany 👍